src/platform/linux/input.cpp

```diff
@@ -1510,7 +1510,7 @@ namespace platf {
     std::stringstream ss;
     ss << std::hex << std::setfill('0');
     for (const auto &ch : str) {
-      ss << ch;
+      ss << static_cast<uint_least32_t>(ch);
```

Comment:
Why are we casting? The `char32_t` is already unsigned.

Reply:
stringstreams don't approve of mixing type `char` and type `char32_t`. (You can try this.) This cast is done to force the stringstream to recognize that this value is a 32-bit integer and to hex it accordingly.

Reply:
```
/home/runner/work/Sunshine/Sunshine/src/platform/linux/input.cpp: In function ‘std::string platf::to_hex(const std::__cxx11::basic_string<char32_t>&)’:
/home/runner/work/Sunshine/Sunshine/src/platform/linux/input.cpp:1513:13: error: use of deleted function ‘std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, char32_t) [with _Traits = std::char_traits<char>]’
 1513 |       ss << ch;
      |             ^~
In file included from /usr/include/c++/10/bits/unique_ptr.h:42,
                 from /usr/include/c++/10/bits/locale_conv.h:41,
                 from /usr/include/c++/10/locale:43,
                 from /usr/include/boost/locale/boundary/facets.hpp:17,
                 from /usr/include/boost/locale/boundary.hpp:12,
                 from /usr/include/boost/locale.hpp:11,
                 from /home/runner/work/Sunshine/Sunshine/src/platform/linux/input.cpp:21:
/usr/include/c++/10/ostream:553:5: note: declared here
  553 |     operator<<(basic_ostream<char, _Traits>&, char32_t) = delete;
      |     ^~~~~~~~
make[2]: *** [tests/CMakeFiles/test_sunshine.dir/build.make:665: tests/CMakeFiles/test_sunshine.dir/__/src/platform/linux/input.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:271: tests/CMakeFiles/test_sunshine.dir/all] Error 2
```